src/main/java/org/spongepowered/api/world/PositionSource.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.world;
+
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.entity.Entity;
+import org.spongepowered.math.vector.Vector3i;
+
+public interface PositionSource {
```

Comment:
Can you document what the point of a PositionSource is? The factory functions are easy to understand, but as mentioned in discord, what the reasoning is for a PositionSource.

Reply:
How should I document it? My only thought on how to describe this is "PositionSource represents something that can have position in world" (I'm kinda bad at this :d) and maybe add  `{@link ParticleTypes#VIBRATION}` because for now it's the only usage of it

Reply:
Well, I guess the point of it is to describe what is the purpose of PositionSource over a Vector3i? 
> In some particle implementations, they can "follow" a source that moves over several ticks such that a singular position is not enough information to follow.

Reply:
The goal should be to describe to someone who is looking at this class for the first time, "What do I do with this? Why do I need this?"

Reply:
Yeah, I understand what the goal is, I'm just not good at expressing thoughts even on my language, not to mention English :D So I added what you suggested